gtest/src/lib.rs

```diff
@@ -28,227 +28,311 @@
+//!
+//! The first one requires payload to be CODEC Encodable, while the second
+//! requires payload implement `AsRef<[u8]>`, that means to be able to represent
+//! as bytes.
+//!
+//! [`Program::send`] uses [`Program::send_bytes`] under the hood with bytes
+//! from `payload.encode()`.
+//!
+//! First message to the initialized program structure is always the init
+//! message.
+//!
+//! ```
+//! let res = program.send_bytes(100001, "INIT MESSAGE");
+//! ```
+//!
+//! ## Processing the result of the program execution
+//!
+//! Any sending functions in the lib returns [`RunResult`] structure.
+//!
+//! It contains the final result of the processing message and others, which
+//! were created during the execution.
+//!
+//! It has 4 main functions:
+//!
+//! - [`RunResult::log`] — returns the reference to the Vec produced to users
+//!   messages. You may assert them as you wish, iterating through them.
+//! - [`RunResult::main_failed`] — returns bool which shows that there was panic
+//!   during the execution of the main message.
+//! - [`RunResult::others_failed`] — returns bool which shows that there was
+//!   panic during the execution of the created messages during the main
+//!   execution. Equals false if no others were called.
+//! - [`RunResult::contains`] — returns bool which shows that logs contain a
+//!   given log. Syntax sugar around `res.log().iter().any(|v| v == arg)`.
+//!
+//! To build a log for assertion you need to use [`Log`] structure with its
+//! builders. All fields here are optional. Assertion with `Log`s from core are
+//! made on the `Some(..)` fields. You will run into panic if you try to set the
+//! already specified field.
+//!
+//! ```
+//! // Constructor for success log.
+//! let log = Log::builder();
+//!
+//! // Constructor for error reply log.
+//! let log = Log::error_builder();
+//!
+//! // Other fields are set optionally by `dest()`, `source()`, `payload()`, `payload_bytes()`.
+//! let log = Log::builder()
+//!     .source(prog.id())
+//!     .dest(100001)
+//!     .payload_bytes("PONG");
+//! ```
+//!
+//! Log also has `From` implementations from `(ID, T)`` and from `(ID_1, ID_2,
+//! T)``, where `ID: Into<ProgramIdWrapper>`, `T: AsRef<[u8]>`.
+//!
+//! ```
+//! let x = Log::builder().dest(5).payload_bytes("A");
+//! let x_from: Log = (5, "A").into();
+//! assert_eq!(x, x_from);
+//!
+//! let y = Log::builder().dest(5).source(15).payload_bytes("A");
+//! let y_from: Log = (15, 5, "A").into();
+//! assert_eq!(y, y_from);
+//! ```
+//!
+//! ## Spending blocks
+//!
+//! You may control time in the system by spending blocks.
+//!
+//! It adds the amount of blocks passed as arguments to the current block of the
+//! system. Same for the timestamp. Note, that for now 1 block in Gear-based
+//! network is 3 sec duration.
```

Comment:
I think it's worth mentioning whether the `send` (other functions) functions spend/do not spend 1 block automatically

Reply:
I have added a note that message processing doesn't spend blocks (in 673b06d).